packages/typespec-azure-core/src/rules/no-generic-numeric.ts

```diff
@@ -0,0 +1,59 @@
+import { Model, Scalar, createRule, paramMessage } from "@typespec/compiler";
+
+const disallowList = new Set(["integer", "numeric", "float", "decimal"]);
+const alternatives = new Map([
+  ["integer", "int32"],
```

Comment:
What are your thoughts about writing a codefix for this?  I am thinking that the 'integer' and 'float' fixes are helpful, but we shoyuld not attempt an automated fix for'numeric' or 'decimal'

Reply:
https://github.com/Azure/typespec-azure/pull/619#discussion_r1566359749